spec/requests/favorites_request_spec.rb

```diff
@@ -0,0 +1,90 @@
+require 'rails_helper'
+
+RSpec.describe "Favorites", type: :request do
+  before do
+    @user = create(:user)
+    sign_in @user
+    get root_path
+  end
+
+  describe "GET /index" do
+    context 'ログインしているとき' do
+      it 'リクエストが成功すること' do
+        get favorites_index_path
+        expect(response.status).to eq 200
+      end
+
+      it 'indexページで表示されること' do
+        cuisine = create(:cuisine, name: "料理1")
+        cuisine2 = create(:cuisine, name: "料理2")
+        favorite = create(:favorite, user_id: @user.id, cuisine_id: cuisine.id)
+        favorite2 = create(:favorite, user_id: @user.id, cuisine_id: cuisine2.id)
+        get favorites_index_path
+        expect(response.body).to include "料理1"
+        expect(response.body).to include "料理2"
+      end
+    end
+
+    context 'ログインしていないとき' do
+      it 'ログインページへリダイレクトすること' do
+        sign_out @user
+        get favorites_index_path
+        expect(response.status).to eq 302
+        expect(response).to redirect_to(new_user_session_path)
+        follow_redirect!
+        expect(response.body).to include "メールアドレス"
+      end
+    end
+  end
+
+  describe "POST /create" do
+    # TODO: テストを追加する
+    context 'ログインしているとき' do
+      # テストがパスしない
+      it '登録できること' do
+        cuisine = create(:cuisine)
+        expect{
+          post cuisine_favorites_path,
+                params: { favorite: attributes_for(:favorite, user_id: @user.id, cuisine_id: cuisine.id)}
+        }.to change{favorite.count}.by(1)
+      end
```

Comment:
- テストがパスしない
- 原因はcontrollerにparamsを定義していないから?

Reply:
エラーメッセージください

Reply:
失礼しました。エラーメッセージは下記になります。
```
Failures:

  1) Favorites POST /create ログインしているとき 登録できること
     Failure/Error:
       expect{
         post cuisine_favorites_path,
               params: { favorite: attributes_for(:favorite, user_id: @user.id, cuisine_id: cuisine.id)}
       }.to change{favorite.count}.by(1)

     NameError:
       undefined local variable or method `favorite' for #<RSpec::ExampleGroups::Favorites::POSTCreate::Nested:0x000055ada3a758b8>
     # ./spec/requests/favorites_request_spec.rb:49:in `block (5 levels) in <top (required)>'
     # ./spec/requests/favorites_request_spec.rb:46:in `block (4 levels) in <top (required)>'
```

Reply:
> undefined local variable or method `favorite' 

favorite定義してないのでエラーになってます。`change{favorite.count}.by(1)`の部分ですね。ローカル変数定義してください。

Reply:
ごめんなさい、現在の自分では`ローカル変数定義する`というのがわからないのですが
```
post する前に
favorite = build(:favorite)
を記述する
```
ということでしょうか?

Reply:
> ということでしょうか?

意味としてはそんな感じです。

因みに、ここってこうじゃないですか？ 単純にtypoな気がします。

```suggestion
      it '登録できること' do
        cuisine = create(:cuisine)
        expect{
          post cuisine_favorites_path,
                params: { favorite: attributes_for(:favorite, user_id: @user.id, cuisine_id: cuisine.id)}
        }.to change{ Favorite.count }.by(1)
      end
```